### PR TITLE
[vendordeps] Use fileName in download

### PR DESCRIPTION
**Pull request by spacey-sooty**
Resolves https://github.com/wpilibsuite/native-utils/issues/196

Also makes testing project use vendordeps plugin

### CODE REVIEW

File: src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java [hunk 1]
```diff
@@ -41,6 +45,22 @@ public void install() throws IOException {
             getLogger().info("Remotely fetching " + filename);
             downloadRemote(dest);
         }
+
+        var destString = dest.toString();
+        String newFilename;
+        try (BufferedReader reader = Files.newBufferedReader(dest)) {
+            newFilename = new GsonBuilder().create().fromJson(reader, JsonDependency.class).fileName;
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java:52]: How will this behave with a malformed vendor JSON? eg one missing the fileName attribute

File: src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java [hunk 2]
```diff
@@ -41,6 +45,22 @@ public void install() throws IOException {
...
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        File file = new File(destString);
+        int lastPathSeperator = dest.toString().lastIndexOf('/');
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java:57]: Typo

```suggestion
        int lastPathSeparator = dest.toString().lastIndexOf('/');
```

File: src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java [hunk 3]
```diff
@@ -41,6 +45,22 @@ public void install() throws IOException {
...
+        boolean didRename = file.renameTo(new File(dest.toString().substring(0, lastPathSeperator + 1) + newFilename));
+        if (didRename == true) {
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java:59]: Can be simplified

```suggestion
        if (didRename) {
```

File: src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java [hunk 4]
```diff
@@ -41,6 +45,22 @@ public void install() throws IOException {
...
+          getLogger().info("File Succesfully Renamed");
+        } else {
+          getLogger().warn("Failed to rename file");
```
Comment (SamCarlberg) [inline src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java:62]: Some information about which file is being renamed and what it was attempted to be renamed to would be helpful

File: src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java (hunk 1 above)
Comment (spacey-sooty) [reply src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java:52]: Crash. I'll add a try catch

Comment (spacey-sooty) [reply src/main/java/edu/wpi/first/nativeutils/vendordeps/VendorDepTask.java:52]: I lie. If it doesn't have fileName file name will be null. Crashing would be caused by Json syntax issues or IO issues

**Review by SamCarlberg — COMMENTED** [review]
LGTM. @ThadHouse?

**Review by ThadHouse — APPROVED** [review]

**Pull request merged**